xlators/cluster/dht/src/dht-rebalance.c

```diff
@@ -4666,40 +4663,43 @@ gf_defrag_status_get(dht_conf_t *conf, dict_t *dict)
         gf_log(THIS->name, GF_LOG_WARNING, "failed to set time-left");
 
 log:
-    switch (defrag->defrag_status) {
-        case GF_DEFRAG_STATUS_NOT_STARTED:
-            status = "not started";
-            break;
-        case GF_DEFRAG_STATUS_STARTED:
-            status = "in progress";
-            break;
-        case GF_DEFRAG_STATUS_STOPPED:
-            status = "stopped";
-            break;
-        case GF_DEFRAG_STATUS_COMPLETE:
-            status = "completed";
-            break;
-        case GF_DEFRAG_STATUS_FAILED:
-            status = "failed";
-            break;
-        default:
-            break;
-    }
+    if (log_status) {
+        switch (defrag->defrag_status) {
+            case GF_DEFRAG_STATUS_NOT_STARTED:
+                status = "not started";
+                break;
+            case GF_DEFRAG_STATUS_STARTED:
+                status = "in progress";
+                break;
+            case GF_DEFRAG_STATUS_STOPPED:
+                status = "stopped";
+                break;
+            case GF_DEFRAG_STATUS_COMPLETE:
+                status = "completed";
+                break;
+            case GF_DEFRAG_STATUS_FAILED:
+                status = "failed";
+                break;
+            default:
+                break;
+        }
 
-    gf_msg(THIS->name, GF_LOG_INFO, 0, DHT_MSG_REBALANCE_STATUS,
-           "Rebalance is %s. Time taken is %.2f secs", status, elapsed);
-    gf_msg(THIS->name, GF_LOG_INFO, 0, DHT_MSG_REBALANCE_STATUS,
-           "Files migrated: %" PRIu64 ", size: %" PRIu64 ", lookups: %" PRIu64
-           ", failures: %" PRIu64
-           ", skipped: "
-           "%" PRIu64,
-           files, size, lookup, failures, skipped);
+        gf_msg(THIS->name, GF_LOG_INFO, 0, DHT_MSG_REBALANCE_STATUS,
```

Comment:
nit: I wouldn't call 'THIS' here and in the below line. Although it doesn't seem to happen from this change, perhaps we can combine both gf_msg() calls to one?

Reply:
Done.